test/e2e/catalog_e2e_test.go

```diff
@@ -138,15 +147,16 @@ var _ = Describe("Catalog represents a store of bundles which OLM can use to ins
 		}
 
 		// Create the initial catalog source
-		createInternalCatalogSource(c, crc, mainCatalogName, globalNS, mainManifests, []apiextensions.CustomResourceDefinition{mainCRD}, []v1alpha1.ClusterServiceVersion{mainCSV})
+		cs, cleanup := createInternalCatalogSource(c, crc, mainCatalogName, globalNS, mainManifests, []apiextensions.CustomResourceDefinition{mainCRD}, []v1alpha1.ClusterServiceVersion{mainCSV})
+		defer cleanup()
```

Comment:
Note: only substantial difference between Per's original PR and this one. I noticed we weren't cleaning up this CatalogSource that's deployed in the global namespace correctly.